go-controller/pkg/util/node_annotations.go

```diff
@@ -1062,3 +1080,21 @@ func makeCIDRs(s ...string) (cidrs []netip.Prefix, err error) {
 	}
 	return cidrs, nil
 }
+
+func remove(strs []string, candidate string) []string {
+	idxsToRemove := make([]int, 0)
+	for i, str := range strs {
+		if str == candidate {
+			idxsToRemove = append(idxsToRemove, i)
+		}
+	}
+	for _, idxToRemove := range idxsToRemove {
+		strs = removeIndex(strs, idxToRemove)
+	}
+	return strs
+}
+
+func removeIndex(strs []string, i int) []string {
+	strs[i] = strs[len(strs)-1]
+	return strs[:len(strs)-1]
+}
```

Comment:
Just wondering if we could make this a generic utility function

Reply:
For sure - done.
I didnt export it from this package though  Premature optimisation :P  I found it hard to find a location for it so i created a new file.  I would love to kill this util pkg in general. Its an anti pattern.